### ISSUE TITLE
New mysterious TestClass warnings with 0.28b2

### ISSUE THREAD

**Issue by ngoldbaum** [cython/cython#2135]
When I compile my cython project with cython 0.28b2, I get warnings like this for every cython file in my project:

```
warning: TestClass:20:10: 'cpdef_method' redeclared
warning: TestClass:31:10: 'cpdef_cname_method' redeclared
```

As far as I can see this is reporting a warning about intermediate code that cython is generating internally rather than anything in my codebase.

Here is an example cython program that seems to be sufficient to trigger these warnings:

```
cimport numpy as np
import numpy as np


cdef class MyClass:
    def foo(self, np.float64_t[:] center, np.float64_t[:,:] points):
        pass
```

To get the warning, simply put this code in a .pyx file and then run `cython` on that file.

**Comment by lesteve** [comment]
Maybe similar to https://github.com/cython/cython/issues/2013? 

The weird TestClass warning was mentioned as well in other issues, e.g. https://github.com/cython/cython/issues/1985#issuecomment-343611428, https://github.com/cython/cython/pull/1869#issuecomment-365231686.

**Comment by ngoldbaum** [comment]
Ah, I'm also seeing #2013 so I'm glad I don't need to report that separately.

It would be nice to get these fixed before the final release of 0.28 as we have our build configured to fail on cython warnings.
